crates/tree-sitter-essence/src/grammar.json

```diff
@@ -0,0 +1,773 @@
+{
```

Comment:
this is auto-generated, right? should it be in the repo? quick google tells me `tree-sitter generate` generates this some of the other files too (like parser.c). if so we shouldn't commit them into the repo, instead call tree-sitter generate during build.

since they are in the repo now, you can delete the files locally and see if tree-sitter generate regenerates them to double check?

Reply:
Everything in the src folder of tree-sitter-essence is auto-generated when `tree-sitter generate` is ran. However, with the folder deleted, the rest of the code won't run unless `tree-sitter generate` is ran again, which will then generate the files again. I assume there's a way to automate this so the command is called during build but I don't know how.

Reply:
to add it as a build-time dependency, add the following to the Cargo.toml file (I think!)

```
[build-dependencies]
tree-sitter-cli = "0.24.4"
```

Reply:
I just added this and tested it out. Thank you!

Reply:
I think tree-sitter isn't in the path, despite it being a build-dependency.

Reply:
Turns out "CARGO_BIN_EXE_" is the way to get this. See https://doc.rust-lang.org/cargo/reference/environment-variables.html

You can try: 

```
    let tree_sitter_path = env::var("CARGO_BIN_EXE_tree-sitter")
        .expect("Failed to locate tree-sitter binary.");

    ... Command::new(tree_sitter_path) ...
```